Scripts/main.js

```diff
@@ -0,0 +1,181 @@
+'use strict'
+
+var Rule = function (value) {
+    this.value = value;
+    this.functionArray = [];
+}
+
+Rule.prototype.setValue = function (value) {
+    this.value = value;
+}
+
+Rule.prototype.isRequired = function () {
+    this.functionArray.push(function (ruleValue) {
+        var isValid = ruleValue ? true : false;
```

Comment:
You can simplify this code: var isValid = !!ruleValue;
But looks like this code works incorrectly with numbers. 0 will marked as invalid value